language/api/analyze_test.py

```diff
@@ -0,0 +1,258 @@
+    assert sentiment['magnitude'] < 1
+
+    result = analyze.analyze_sentiment(
+        'cheerio, mate - I greatly admire the pallor of your visage, and your '
+        'angle of repose leaves little room for improvement.')
+
+    sentiment = result['documentSentiment']
+    assert sentiment['polarity'] > 0
+    assert sentiment['magnitude'] < 1
+
+
+def test_analyze_syntax(capsys):
+    result = analyze.analyze_syntax(textwrap.dedent(u'''\
+        Keep away from people who try to belittle your ambitions. Small people
+        always do that, but the really great make you feel that you, too, can
+        become great.
+        - Mark Twain'''))
+
+    assert len(result['tokens'])
+    first_token = result['tokens'][0]
+    assert first_token['text']['content'] == 'Keep'
+    assert first_token['partOfSpeech']['tag'] == 'VERB'
+    assert len(result['sentences']) > 1
+    assert result['language'] == 'en'
+
+
+def test_analyze_syntax_utf8():
+    """Demonstrate the interpretation of the offsets when encoding=utf8.
+
+    UTF8 is a variable-length encoding, where each character is at least 8
+    bits. The offsets we get should be the index of the first byte of the
+    character.
+    """
+    test_string = u'a \u00e3 \u0201 \U0001f636 b'
+    byte_array = test_string.encode('utf8')
+    result = analyze.analyze_syntax(test_string, encoding='UTF8')
+    tokens = result['tokens']
+
+    assert tokens[0]['text']['content'] == 'a'
+    offset = tokens[0]['text'].get('beginOffset', 0)
+    assert (byte_array[offset:offset+1].decode('utf8') ==
+            tokens[0]['text']['content'])
+
+    assert tokens[1]['text']['content'] == u'\u00e3'
+    offset = tokens[1]['text'].get('beginOffset', 0)
+    assert (byte_array[offset:offset+2].decode('utf8') ==
+            tokens[1]['text']['content'])
+
+    assert tokens[2]['text']['content'] == u'\u0201'
+    offset = tokens[2]['text'].get('beginOffset', 0)
+    assert (byte_array[offset:offset+2].decode('utf8') ==
+            tokens[2]['text']['content'])
+
+    assert tokens[3]['text']['content'] == u'\U0001f636'
+    offset = tokens[3]['text'].get('beginOffset', 0)
+    assert (byte_array[offset:offset+4].decode('utf8') ==
+            tokens[3]['text']['content'])
+
+    # This demonstrates that the offset takes into account the variable-length
+    # characters before the target token.
+    assert tokens[4]['text']['content'] == u'b'
+    offset = tokens[4]['text'].get('beginOffset', 0)
+    # 'b' is only one byte long
+    assert (byte_array[offset:offset+1].decode('utf8') ==
+            tokens[4]['text']['content'])
+
+
+def test_analyze_syntax_utf16():
+    """Demonstrate the interpretation of the offsets when encoding=utf16.
+
+    UTF16 is a variable-length encoding, where each character is at least 16
+    bits. The returned offsets will be the index of the first 2-byte character
+    of the token.
+    """
+    test_string = u'a \u00e3 \u0201 \U0001f636 b'
```

Comment:
It does but I wonder if it's more or less clear if you actually use a printable character and then decode it into bytes. Not sure, food for thought mostly.